<launch>
    
    <param name="robot_name" value="pr2" />
    <param name="step_size" value="0.02" />
    <param name="draw_points" value="false" />

    <node name="collision_check_service" pkg="lightning" type="collision_check_service" output="screen">
        <param name="num_threads" value="4" />
    </node>

    <node name="path_shortcutter_service" pkg="lightning" type="path_shortcutter" output="screen">
        <param name="num_iterations" value="200" />
        <param name="ignore_fraction" value="0.1" />
    </node>

    <!-- load planning -->
    <include file="$(find lightning)/launch/planners.launch"/>

    <node name="rr_node" pkg="lightning" type="RR_action_server.py" output="screen">
        <param name="num_rr_planners" value="2" />
        <param name="path_library_path_node_size" value="8" />
        <param name="path_library_sg_node_size" value="50" />
        <param name="num_paths_to_collision_check" value="4" />
        <param name="dtw_distance" value="5.0" />
        <param name="path_library_dir" value="$(find lightning)/src/pathlib/paths" />
    </node>

    <node name="pfs_node" pkg="lightning" type="PFS_action_server.py" output="screen">
    </node>

    <node name="lightning" pkg="lightning" type="lightning.py" output="screen">
        <param name="store_paths" value="true" />
        <param name="use_RR" value="true" />
        <param name="use_PFS" value="true" />
    </node>

</launch>
